src/puppetlabs/trapperkeeper/authorization/rules.clj

```diff
@@ -125,67 +121,90 @@
             (some (get rule-params k)
                   (flatten [(get request-params (name k))])))))
 
-(schema/defn match? :- RuleMatch
+(schema/defn sort-rules :- [Rule]
+  "Sorts the rules based on their :sort-order, and then their :name if they
+   have the same sort order value."
+  [rules :- [Rule]]
+  (sort-by (juxt :sort-order :name) rules))
+
+(schema/defn requestor :- schema/Str
+  "Returns a string that identifies the source of the request containing
+   at least the IP address and the hostname if available."
+  [request :- ring/Request]
+  (if-let [name (ring/authorized-name request)]
+    (format "%s(%s)" name (:remote-addr request))
+    (str (:remote-addr request))))
```

Comment:
If it was all just for this, I'd probably instead have done something like:

~~~~clj
(let [name (ring/authorized-name request)]
  (if (empty? name)
     (str (:remote-addr request))
     (format "%s(%s)" name (:remote-addr request))))
~~~~

Not great having the extra level of nesting but I do think it is valuable to avoid passing around `nil` when an empty string would generally work fine.

Again, I'm not too concerned about this, though.  Would like to get @dankreek's opinion.